### PR TITLE
Carolina testing workflow use Dakota 6.18

### PR DESCRIPTION
**Pull request by andreas-el**
Update GitHub testing workflow to use Dakota 6.18

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 name: Python package
 
 env:
-  DAKOTA_VERSION: 6.17.0
-
+  DAKOTA_VERSION: 6.18.0
+  DAKOTA_FILENAME: dakota-6.18.0-public-src-cli
```
Comment (sregales-TNO) [inline .github/workflows/testing.yml]: Is it possible to have one change when ever the version chnages? Here it is still two `DAKOTA_VERSION` and `DAKOTA_FILENAME`

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -47,7 +47,6 @@ jobs:
         cd build
         cmake \
           -DBUILD_SHARED_LIBS=ON \
-          -DDAKOTA_PYTHON=ON \
```
Comment (sregales-TNO) [inline .github/workflows/testing.yml:50]: other than it runs without it, do you have any evidence that it's unused?

Comment (andreas-el) [reply .github/workflows/testing.yml:50]: I don't, but @verveerpj removed this option a shortly before summer.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (andreas-el) [reply .github/workflows/testing.yml]: I'll fix this. 👍

**Review by sregales-TNO — APPROVED** [review]
LGTM

**Pull request merged**